lib/logstash/outputs/mongodb.rb

```diff
@@ -38,20 +41,18 @@ def register
     @db = conn.use(@database)
   end # def register
 
-  public
   def receive(event)
-    
-
     begin
-      if @isodate
-        # the mongodb driver wants time values as a ruby Time object.
-        # set the @timestamp value of the document to a ruby Time object, then.
-        document = event.to_hash
-      else
-        document = event.to_hash.merge("@timestamp" => event["@timestamp"].to_json)
+      # Our timestamp object now has a to_bson method, using it here
+      # why not use to_hash_with_metadata?
+      # {}.merge(other) so we don't taint the event hash innards
+      document = {}.merge(event.to_hash)
```

Comment:
why using merge with the event data? would not be more clear to use ```event.to_hash.clone```? I wonder about your way of thinking here? are the implications for it?